portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -17,24 +17,37 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Date;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import java.util.Date;
-import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String json = getInformationInJson();
+        // Fetch the comment from the datastore
+        PreparedQuery comments = datastore.prepare(new Query("Comment"));
+
+        // Store the contents of those comments in a list
+        List<String> commentList = new ArrayList<>();
+        for (Entity entity : comments.asIterable()) {
+            commentList.add((String) entity.getProperty("content"));
+        }
+
+        // Respond the comments as json form
         response.setContentType("application/json");
-        response.getWriter().println(json);
+        response.getWriter().println(new Gson().toJson(commentList));
```

Comment:
Actually, I would suggestion make Gson static too.
Maybe not need to make class scope static, function scope static is enough.

Reply:
Sorry, I don't quit understand. Do you mean:
`static gson = new Gson();`
`response.getWriter().println(gson.toJson(commentList));` 
But, will it be redundant, cause the gson is only used once here in the whole class?

Reply:
You could write
```
static final Gson GSON = new Gson();
response.getWriter().println(GSON.toJson(commentList));
```

used once in code.
But basically it means that you need to new one Gson object for every Get request.